corehq/apps/sso/forms.py

```diff
@@ -618,6 +643,57 @@ def get_primary_fields(self):
             ),
         ]
 
+    def get_remote_user_management_fields(self):
+        def mask_api(s):
+            if len(s) <= 3:
+                return s
+            else:
+                return s[:3] + '*' * (len(s) - 3)
```

Comment:
Up to you. I tend to avoid using nested functions, but I think I tend to believe they're used for generally simple functions that are tightly coupled to the parent function. Because we see the potential for generic re-use here, I think we can probably make this its own stand-alone function

Reply:
I agree with @mjriley. I think best practice here would be to create a utility called `get_masked_string(string, mask_character=None, reveal_length=3)` that takes in a string, optional `mask_character` defaults to '*' if `None` is provided. This can live in `corehq.util.strings` and a quick test can be written for it. Seems like a utility we can use elsewhere.

Reply:
Thanks Matt and Biyeun! The utility function and the test is in  703a0eca330195505c1c7090254c86d2fab01ff1